tools/CheckH5.py

```diff
@@ -0,0 +1,207 @@
+                    for index in np.ndindex(test_data.shape):
+                        # It is recommended to include subTest commands once
+                        # python3 can be required:
+                        # with self.unit_test.subTest(index=index):
+                        self.unit_test.assertTrue(
+                            self.approx(test_data[index],
+                                        expected_data[index]),
+                            msg="Compare " + str(test_data[index]) + " to " +
+                            str(expected_data[index]))
+                else:
+                    self.unit_test.assertEqual(test_data, expected_data)
+            else:
+                if (test_data.dtype == np.dtype(np.floating)
+                        or test_data.dtype == np.dtype(np.complexfloating)):
+                    for index in np.ndindex(test_data.shape):
+                        # It is recommended to include subTest commands once
+                        # python3 can be required:
+                        # with self.unit_test.subTest(index=index):
+                        self.unit_test.assertTrue(
+                            self.approx(test_data[index], 0.0))
+                else:
+                    self.assertTrue(
+                        False,
+                        msg="cannot test non-numeric data without an"
+                        " expected data set to compare against")
+        elif isinstance(h5_file[test_entity], h5py.Group):
+            test_keys = set(h5_file[self.test_entity].keys())
+            checks_passed = True
+            if self.expected_h5_entity is not None:
+                expected_keys = set(h5_file[self.expected_entity].keys())
+                keys_difference = test_keys ^ expected_keys
+                self.unit_test.assertEqual(
+                    keys_difference, {},
+                    "test and expected h5 groups must have identical" +
+                    "subgroups and data members. Found differences: " +
+                    str(keys_difference) + "\nin comparing group " +
+                    test_entity + " to group " + expected_entity)
+                for key in test_keys:
+                    checks_passed = checks_passed and self.check_h5_file(
+                        h5_file, test_entity + "/" + key,
+                        expected_entity + "/" + key)
+            else:
+                for key in test_keys:
+                    checks_passed = checks_passed and self.check_h5_file(
+                        h5_file, test_entity + "/" + key, None)
+            return checks_passed
+
+    # Apply the h5 check to every h5 file within the `run_directory`.
+    # If neither the expected nor test objects are present in an h5 file, that
+    # file is skipped. However, if none of the h5 files have the requested
+    # datasets, the test fails.
+    # It is not supported for the `run_directory` to have h5 files from
```

Comment:
sorry what do you mean here? Is this going to cause problems for tests run in parallel? I'm worried that this script will open a file and start looking through it, then a parallel test deletes the file under it because the other test "owns" the file. This should also be documented